study/src/test/java/di/stage4/annotations/ClassPathScanner.java

```diff
@@ -1,10 +1,13 @@
 package di.stage4.annotations;
 
 import java.util.Set;
+import org.reflections.Reflections;
+import org.reflections.scanners.Scanners;
 
 public class ClassPathScanner {
 
     public static Set<Class<?>> getAllClassesInPackage(final String packageName) {
-        return null;
+        Reflections reflections = new Reflections(packageName, Scanners.SubTypes.filterResultsBy(filter -> true));
```

Comment:
`new Reflection(packageName)`과 어떤 점이 다른가요??

Reply:
- `new Reflection(packageName)`은 지정된 패키지 내 모든 클래스를 스캔 하고, 필터링을 하지 않습니다.
- `new Reflections(packageName, Scanners.SubTypes.filterResultsBy(filter -> true))`는 Scanners.SubTypes를 사용하여 서브타입만 스캔하며, 필터링을 할 수 있습니다.

답변이 질문의 의도에 맞나요? ?

Reply:
그렇다면 어차피 모든 클래스를 스캔할 예정인데 굳이 뒤에 Scanner가 필요한 이유는 뭘까요??

Reply:
> Scanners must be configured in order to be queried, otherwise an empty result is returned.

Reflection 공식 문서에 위와 같이 나와 있었습니다. 😸
Scanners를 설정하지 않으면 실제로는 클래스를 스캔하지 않는다고 합니다.

Reply:
맞아요~~~
스캐너 설정이 없으면 빈 결과가 나오더라고요
저는 거기서 엄청 헤맸답니다 😹